components/feature/prompts/src/main/java/mozilla/components/feature/prompts/login/LoginSelectBar.kt

```diff
@@ -0,0 +1,138 @@
+                headerTextStyle = textStyle
+            }
+        }
+    }
+
+    override var listener: LoginPickerView.Listener? = null
+
+    override fun showPicker(list: List<Login>) {
+        tryInflate().also {
+            listAdapter.submitList(list)
+            loginPickerView?.isVisible = true
+        }
+    }
+
+    override fun hidePicker() {
+        this.isVisible = false
+        loginsList?.isVisible = false
+        listAdapter.submitList(mutableListOf())
+        manageLoginsButton?.isVisible = false
+        toggleSavedLoginsHeader(shouldExpand = false)
+    }
+
+    override fun asView(): View {
+        return super.asView()
+    }
+
+    private var loginPickerView: View? = null
+    private var loginsList: RecyclerView? = null
+    private var manageLoginsButton: AppCompatTextView? = null
+    private var savedLoginsHeader: AppCompatTextView? = null
+    private var expandArrowHead: AppCompatImageView? = null
+
+    private var listAdapter = BasicLoginAdapter {
+        listener?.onLoginSelected(it)
+    }
+
+    override fun tryInflate(): Boolean {
+        return if (loginPickerView == null) {
+            loginPickerView =
+                View.inflate(context, R.layout.mozac_feature_login_multiselect_view, this)
+            bindViews()
+            true
+        } else {
+            false
+        }
+    }
+
+    private fun bindViews() {
+        manageLoginsButton = findViewById<AppCompatTextView>(R.id.manage_logins).apply {
+            setOnClickListener {
+                listener?.onManageLogins()
+            }
+        }
+        loginsList = findViewById(R.id.logins_list)
+        savedLoginsHeader = findViewById<AppCompatTextView>(R.id.saved_logins_header).apply {
+            headerTextStyle?.let {
+                TextViewCompat.setTextAppearance(this, it)
+                currentTextColor.let {
+                    TextViewCompat.setCompoundDrawableTintList(this, ColorStateList.valueOf(it))
+                }
+            }
+            setOnClickListener {
+                toggleSavedLoginsHeader(shouldExpand = loginsList?.isVisible != true)
+            }
+        }
+        expandArrowHead =
+            findViewById<AppCompatImageView>(R.id.mozac_feature_login_multiselect_expand).apply {
+                savedLoginsHeader?.currentTextColor?.let {
+                    ImageViewCompat.setImageTintList(this, ColorStateList.valueOf(it))
+                }
+            }
+        loginsList?.apply {
+            layoutManager = LinearLayoutManager(context, RecyclerView.VERTICAL, false)
+            adapter = listAdapter
+        }
+    }
+
+    private fun toggleSavedLoginsHeader(shouldExpand: Boolean) {
+        if (shouldExpand) {
+            loginsList?.isVisible = true
+            manageLoginsButton?.isVisible = true
+            loginPickerView?.hideKeyboard()
+            expandArrowHead?.rotation = ROTATE_UPSIDE_DOWN
+        } else {
+            expandArrowHead?.rotation = 0F
```

Comment:
ROTATE_DOWNSIDE_UP? 🙃

Reply:
good thing i didnt go with FLIPPIDITY_DIPPITY

Reply:
⅄┴IԀԀIp‾⅄pIԀԀI˥Ⅎ would work but aapt might get mad at you.